pkg/controller/clusterinstallation/canary.go

```diff
@@ -0,0 +1,33 @@
+package clusterinstallation
+
+import (
+	"github.com/go-logr/logr"
+
+	mattermostv1alpha1 "github.com/mattermost/mattermost-operator/pkg/apis/mattermost/v1alpha1"
+)
+
+func (r *ReconcileClusterInstallation) checkCanary(mattermost *mattermostv1alpha1.ClusterInstallation, reqLogger logr.Logger) error {
+	if mattermost.Spec.Canary.Enable {
+		ingressAnnotations := map[string]string{
+			"kubernetes.io/ingress.class":                  "nginx",
+			"nginx.ingress.kubernetes.io/canary":           "true",
+			"nginx.ingress.kubernetes.io/canary-by-cookie": "canary",
+		}
+		reqLogger = reqLogger.WithValues("Reconcile", "mattermost")
```

Comment:
nit: we normally set this logger name right at the top of the function, but really not a big deal.